src/usePromiseWithInterval.ts

```diff
@@ -0,0 +1,35 @@
+import { MutableRefObject, useCallback, useEffect, useRef } from "react";
+import { PromiseLoader, UsePromiseWithInterval } from "./types";
+import { usePromise } from "./usePromiseMatcher";
+
+export const usePromiseWithInterval = <T, Args extends any[], E = string>(
+    loaderFn: PromiseLoader<T, Args>,
+    interval: number,
+): UsePromiseWithInterval<T, E, Args> => {
+    const [result, load, reset] = usePromise<T, Args, E>(loaderFn);
+
+    const timer: MutableRefObject<any | null> = useRef(null);
+
+    const start = useCallback(
+        (...args: Args) => {
+            timer.current = setTimeout(async function tick() {
+                await load(...args);
```

Comment:
We might want to be able to choose whether `setTimeout` or `setInterval` is used here. It's possible that someone might want to send some information to the backend in fixed intervals, then this solution will not work, because the next request will be sent after the previous one finishes. Also, let's imagine a situation where we're loading some short-lived data every N seconds, if one of the requests took long time, we don't want to keep the old data and wait the whole interval for the next one. Not sure yet how to name this argument, maybe sth like `pollingStrategy`?

Reply:
I'd think about splitting this feature into 2 hooks, one that uses `setTimeout`, and the second one with `setInterval`. That's because conditional return of these timers might be tricky to describe as a single type in Typescript.

Also, that would mean that current `usePromiseWithInterval` hook shall be renamed to `usePromiseWithTimeout`. That, on the other hand, could be misleading as you could probably connect the `timeout` with single execution after certain amount of time. 

So maybe: `usePromiseWithTimeoutPolling` and `usePromiseWithIntervalPolling`? 🤔

Reply:
Actually, this would require also some refactoring related to the file structure, as there would be more files in the `src` directory. I can create an alternative PR and this one could be closed then.

Reply:
I'd stick to one hook and just return our own interface containing a cancelling function and that's it. We just have to think about good naming for polling strategy, I'll have a look at how is it called in RxJS as there is a similar issue. Let's merge it as it is now though.